src/system.jl

```diff
@@ -0,0 +1,302 @@
+the service requirements for the zone.
+"""
+struct Zone
+    number::Int64
+    reg::Float64
+    spin::Float64
+    sup_on::Float64
+    sup_off::Float64
+end
+
+###### Static Component Types ######
+
+abstract type StaticComponent end
+
+Base.length(components::StaticComponent) = length(getfield(components, 1))
+# define interfaces? Iterator? Table?
+
+"""
+    Generators(
+        name::Vector{Int}
+        zone::Vector{Int}
+        startup_cost::Vector{Float64}
+        shutdown_cost::Vector{Float64}
+        no_load_cost::Vector{Float64}
+        time_at_status::Vector{Float64}
+        min_uptime::Vector{Float64}
+        min_downtime::Vector{Float64}
+        ramp_up::Vector{Float64}
+        ramp_down::Vector{Float64}
+        initial_gen::Vector{Float64}
+        technology::Vector{Symbol}
+    )
+
+Type for static generator component attributes (i.e. things that describe a generator that
+are not time series data).
+"""
+struct Generators <: StaticComponent
+    name::Vector{Int}
+    zone::Vector{Int}
+    startup_cost::Vector{Float64}
+    shutdown_cost::Vector{Float64}
+    no_load_cost::Vector{Float64}
+    time_at_status::Vector{Float64}
+    min_uptime::Vector{Float64}
+    min_downtime::Vector{Float64}
+    ramp_up::Vector{Float64}
+    ramp_down::Vector{Float64}
+    initial_gen::Vector{Float64} # this one changes in RT with _update_system_generation - but is that necessary - could be a mutable time series?
+    technology::Vector{Symbol}
+end
+
+"""
+    gens_per_zone(gens::Generators)
+
+Returns a `Dict` with  keys of `Zone` numbers and values of generator names in that zone.
+"""
+function gens_per_zone(gens::Generators)
+    gens_per_zone = Dict{Int, Vector{Int}}()
+    for (name, zone) in zip(gens.name, gens.zone)
+        if haskey(gens_per_zone, zone)
+            push!(gens_per_zone[zone], name)
+        else
+            gens_per_zone[zone] = [name]
+        end
+    end
+    return gens_per_zone
+end
+
+"""
+    Buses(name, base_voltage)
+
+Type for static bus component attributes.
+"""
+struct Buses <: StaticComponent
+    name::Vector{String}
+    base_voltage::Vector{Float64}
+end
+
+"""
+    Branches(
+        name::Vector{String}
+        to_bus::Vector{String}
+        from_bus::Vector{String}
+        rate_a::Vector{Float64}
+        rate_b::Vector{Float64}
+        is_monitored::Vector{Bool}
+        break_points::Vector{Tuple{Vararg{Float64}}}
+        penalties::Vector{Tuple{Vararg{Float64}}}
+    )
+
+Type for static branch component attributes.  Branches may have between 0 and 2 break points
+which is why the `break_points` and `penalties` fields contain variable length `Tuple`s.
+"""
+struct Branches <: StaticComponent
+    name::Vector{String}
+    to_bus::Vector{String}
+    from_bus::Vector{String}
+    rate_a::Vector{Float64}
+    rate_b::Vector{Float64}
+    is_monitored::Vector{Bool}
+    break_points::Vector{Tuple{Vararg{Float64}}} # variable length (0, 1, 2)
```

Comment:
do we need the break_point values or just the number of break points (in which case this can be an integer)?